inc/plugins/namespace.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Sets DID-less plugin slug as active.
+ *
+ * @package FAIR
+ */
+
+namespace FAIR\Plugins;
+
+use function FAIR\Updater\get_packages;
+
+/**
+ * Bootstrap
+ *
+ * @return void
+ */
+function bootstrap() {
+	add_filter( 'option_active_plugins', __NAMESPACE__ . '\\set_as_active' );
+	add_filter( 'wp_admin_notice_markup', __NAMESPACE__ . '\\hide_notice', 10, 3 );
+
+	// just for testing.
+	if ( ! function_exists( 'is_plugin_active' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+	}
+	if ( is_plugin_active( 'git-updater/git-updater.php' ) ) {
+		wp_admin_notice( 'Git Updater is active' );
+	}
+}
+
+/**
+ * Set FAIR plugins as active using DID-less slug.
+ *
+ * @param  array $active_plugins Array of active plugins.
+ *
+ * @return array
+ */
+function set_as_active( $active_plugins ) {
+	remove_filter( 'option_active_plugins', __NAMESPACE__ . '\\set_as_active' );
+	$packages = get_packages();
+	$plugins = $packages['plugins'] ?? [];
+	foreach ( $plugins as $plugin ) {
+		if ( is_plugin_active( plugin_basename( $plugin ) ) ) {
+			$plugins[] = get_didless_slug( $plugin );
+		}
+	}
+	$active_plugins = array_map( 'plugin_basename', array_merge( $active_plugins, $plugins ) );
+
+	return array_unique( $active_plugins );
+}
+
+/**
+ * Return shortened DID withou `did:plc|web'.
+ *
+ * @param  string $did Full DID.
+ *
+ * @return string|void
+ */
+function get_short_did( $did ) {
+	if ( ! empty( $did ) ) {
+		if ( str_contains( $did, ':' ) ) {
```

Comment:
I'll be looking at what our minimum WP version should be bumped to, and opening a PR as soon as I can, so this doesn't need to be changed to `strpos()`.

Still, with a current minimum PHP of 7.4 and WP of 5.4, bad Andy for using this function. *hits with ruler*

Reply:
str_contains() is in a shim in WP5.9

Reply:
Yep, and our current minimum is 5.4. We also use `wp_admin_notice()`, which is WP 6.4, so I know we'll need to at least bump our minimum to 6.4. I'll work out if there are any other uses that need us to bump above 6.4.

Reply:
Besides Ryan uses `str_starts_with()` in `parse_did()` 😆